src/platform/libretro/libretro.c

```diff
@@ -63,6 +63,27 @@ static void _reloadSettings(void) {
 	};
 
 	struct retro_variable var;
+	enum GBModel model;
+	const char* modelName;
+
+	var.key = "mgba_model";
+	var.value = 0;
+	if (environCallback(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value) {
+		if (strcmp(var.value, "Game Boy") == 0) {
+			model = GB_MODEL_DMG;
+		} else if (strcmp(var.value, "Super Game Boy") == 0) {
+			model = GB_MODEL_SGB;
+		} else if (strcmp(var.value, "Game Boy Color") == 0) {
+			model = GB_MODEL_CGB;
+		} else {
+			model = GB_MODEL_AUTODETECT;
+		}
+
+		modelName = GBModelToName(model);
+		mCoreConfigSetDefaultValue(&core->config, "gb.model", modelName);
```

Comment:
These are three different values for a reason. Unifying them would cause more problems than it would fix.

Reply:
Can you elaborate more on what problems unifying them could cause? I don't want to clutter up the menu too much, so it seems nice having one switch to force the model regardless of whether the game is DMG, CGB, or SGB enhanced.

Since RetroArch supports per-game config files, users can select a model for a specific game and use autodetection for everything else.

Reply:
If it's per-game it's not a big deal. Mostly the problem lies in having the default be:

- DMG/SGB for CGB games will break a lot of things
- CGB/AGB for DMG games can cause incompatibilities at times (only a handful of games though)
- GCB/AGB for SGB games breaks SGB features

which is why I made it be separate categories. But basically there's no good way to control all three at once without something going wrong along the way.